### PR TITLE
OG image cropping

### PR DESCRIPTION
**Pull request by Newnab**
If appsettings contains relevant crop sizes, force any og image to be the best possible dimensions. Adjacent PR incoming to add the sizes to our site boilerplate.

### CODE REVIEW

File: MetaTags/Services/MetaTagsService.cs [hunk 1]
```diff
@@ -124,13 +124,13 @@ private string GetHostUrl()
...
-            {
-                return string.Empty;
+            if (string.IsNullOrWhiteSpace(path))
+            {
+                return string.Empty;
             }
 
             var imageUrl = _mediaFileStore.MapPathToPublicUrl(path);
-            return imageUrl.StartsWith("http") ? imageUrl : $"{GetHostUrl()}{imageUrl}";
+            return imageUrl.StartsWith("http") ? imageUrl : $"{GetHostUrl()}{imageUrl}?width=1200&height=630&rmode=crop";
```
Comment (peterkeating) [inline MetaTags/Services/MetaTagsService.cs]: We'll need to make sure that `630` is added to the app settings in the site boilerplate.

https://github.com/EtchUK/Etch.OrchardCore.SiteBoilerplate/blob/main/Content/src/Etch.OrchardCore.SiteBoilerplate/appsettings.json

Comment (peterkeating) [reply MetaTags/Services/MetaTagsService.cs]: Also not a fan of 1200 & 630 being magic numbers, can we move them to constants.

**Review by peterkeating — CHANGES_REQUESTED** [review]

File: MetaTags/Services/MetaTagsService.cs (hunk 1 above)
Comment (Newnab) [reply MetaTags/Services/MetaTagsService.cs]: Committed some changes. Is that what you meant?

Comment (peterkeating) [reply MetaTags/Services/MetaTagsService.cs]: Not quite. You need to add the following to the top of the class.

```
private const int ImageCropHeight = 630;
private const int ImageCropWidth = 1200;
```

Then use those variables 👍

**Review by peterkeating — APPROVED** [review]

**Pull request merged**